src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const grouOfKey = item[key];
```

Comment:
There is a typo in the variable name `grouOfKey`. It should be `groupOfKey` to correctly represent its purpose and to avoid potential confusion.